README.md

```diff
@@ -1,137 +1,145 @@
 }
 let grades = Grades()
 ```
 
-Next we create a pool of connections to PostgreSQL:
-
+3. Create a pool of connections to PostgreSQL:
 ```swift
-let pool = PostgreSQLConnection.createPool(host: "localhost", port: 5432, options: [.userName("username"), .password("password")], poolOptions: ConnectionPoolOptions(initialCapacity: 10, maxCapacity: 50, timeout: 10000))
+let pool = PostgreSQLConnection.createPool(host: "localhost", port: 5432, options: [.databaseName("school")], poolOptions: ConnectionPoolOptions(initialCapacity: 10, maxCapacity: 50, timeout: 10000))
 ```
 
-Every time we need a connection, we get it from the pool:
-
-```swift
-if let connection = pool.getConnection() {
-   // Build and execute your query here.
-}
-else {
-   print("Error: failed to get a connection.")
-}
-```
-
-Now lets build the query. Suppose we want to retrieve the average grades for courses with average greater than 90, and sort the results by the average ascending. Here is the SQL query we need to build:
-
-```SQL
-SELECT course, ROUND(AVG(grade), 1) AS "average" FROM grades
-GROUP BY course
-HAVING AVG(grade) > 90
-ORDER BY AVG(grade) ASC
-```
-Note, that we also round the average grades and alias this column in the result as "average".
-
-Here is how to create such query using Swift-Kuery:
-
-```swift
-let query = Select(grades.course, round(avg(grades.grade), to: 1).as("average"), from: grades)
-            .group(by: grades.course)
-            .having(avg(grades.grade) > 90)
-            .order(by: .ASC(avg(grades.grade)))
+4. Create some example students:
+```Swift
+let students: [[Any]] = [[0, "computing", 92], [1, "physics", 75], [2, "history", 83]]
 ```
-As you can see, it is very similar to the SQL query syntax.
 
-Now we execute the created query on our PostgreSQL connection:
+5. Connect to database and perform an SQL query:
 
 ```swift
-guard let connection = pool.getConnection() else {
-   // Error
-}
-
-connection.execute(query: query) { queryResult in
-  if let resultSet = queryResult.asResultSet {
-    for title in resultSet.titles {
-      // The column names of the result.
-    }
-    for row in resultSet.rows {
-      for value in row {
-        ...
-      }
+if let connection = pool.getConnection() {
+    let insertQuery = Insert(into: grades, rows: students)
+    connection.execute(query: insertQuery) { insertResult in
+        connection.execute(query: Select(from: grades)) { selectResult in
+            if let resultSet = selectResult.asResultSet {
+                for row in resultSet.rows {
+                    print("Student \(row[0] ?? ""), studying \(row[1] ?? ""), scored \(row[2] ?? "")")
+                }
+            }
+        }
     }
-  }
-  else if let queryError = result.asError {
-      // Something went wrong.
-  }
 }
 ```
-The expected result is:
 
-```
-course     average
-chemistry  92.0
-history    96.0
-```
+Save this file, run `swift build` and then run the created executable module.
```

Comment:
This is too squashed under the code section, I also think it would be clearer as...

5. Save the `main.swift` file. Run `swift build` to build the executable.

6. Run the executable `.build/debug/<yourPackageName>`.